src/textual/widgets/_tree.py

```diff
@@ -307,6 +307,7 @@ def set_label(self, label: TextType) -> None:
         self._updates += 1
         text_label = self._tree.process_label(label)
         self._label = text_label
+        self._tree.call_later(self._tree._refresh_node, self)
```

Comment:
I never know whether I should use `call_later`, `call_soon`, or the other one.
How do you decide?

Reply:
Did you mean `call_next` rather than `call_soon`? I generally go with the rule that `call_next` is the one to go with unless it's obvious otherwise because jumping the queue should be an exceptional thing to do.

Reply:
That makes sense, thanks.